packages/mask/src/plugins/WalletService/services/wallet/index.ts

```diff
@@ -215,7 +215,14 @@ export async function exportPrivateKey(address: string, unverifiedPassword?: str
 
 export async function exportKeyStoreJSON(address: string, unverifiedPassword?: string) {
     if (unverifiedPassword) await password.verifyPasswordRequired(unverifiedPassword)
-    const password_ = await password.INTERNAL_getPasswordRequired()
+    let password_: string = ''
+    try {
+        password_ = await password.INTERNAL_getPasswordRequired()
+    } catch {
+        if (!unverifiedPassword) throw new Error('No password set yet or expired.')
```

Comment:
Use `hasSecret` or `hasSafeSecret` to detect whether a password was set before.

Reply:
`hasSecret` and `hasSafeSecret ` is not for password set detection.